### ISSUE TITLE
Migrating to ubuntu 20.04/noetic

### ISSUE THREAD

**Issue by dulex123** [HKUST-Aerial-Robotics/Fast-Planner#85]
I am trying to build master branch in Noetic, apart from installing ros-noetic-pcl* and changing std=c++11 flags to std=c++14 in all cmake files (PCL 1.10 requires it) everything compiles. 

I **can run both** roslaunch plan_manage rviz.launch and roslaunch plan_manage kino_replan.launch

**But when I issue navgoal it crashes without much information:**

[fast_planner_node-1] process has died [pid 23197, exit code -11, cmd ~/catkin_ws/devel/lib/plan_manage/fast_planner_node /odom_world:=/state_ukf/odom /sdf_map/odom:=/state_ukf/odom /sdf_map/cloud:=/pcl_render_node/cloud /sdf_map/pose:=/pcl_render_node/camera_pose /sdf_map/depth:=/pcl_render_node/depth __name:=fast_planner_node __log:=~/.ros/log/c518fd5c-3f9b-11eb-9def-dd79eedcd81f/fast_planner_node-1.log].
log file: ~/.ros/log/c518fd5c-3f9b-11eb-9def-dd79eedcd81f/fast_planner_node-1*.log

**I have ran the second command with --log so I have output of fast_planner_node but it doesnt say much:**

hit: 0.619039
miss: -0.619039
min log: -1.99243
max: 2.19722
thresh log: 1.38629
origin_: -20 -10  -1
map size: 40 20  5
[FSM]: from INIT to WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
[FSM]: state: WAIT_TARGET
Triggered!
[TRIG]: from WAIT_TARGET to GEN_NEW_TRAJ
[kino replan]: -----------------------
start:         -19           0 1.51788e-18, 0 0 0, 0 0 0
goal:-9.50174 -1.04982        1, 0 0 0
open set empty, no path!
use node num: 1
iter num: 1
[kino replan]: kinodynamic search fail!
reach horizon
[kino replan]: retry search success.

**Rviz minor issue: Simulation-robot marker is not showing it says: status:error for frame /world fixed frame [world] does not exist**

**Comment by WANGSSSSSSS** [comment]
i meet this error too, have you solve this problem yet?

**Comment by yjhdhr** [comment]
hello, the same problem ...

**Comment by yjhdhr** [comment]
> i meet this error too, have you solve this problem yet?

HI, do you know how to fix it?

**Comment by surfii3z** [comment]
Same here, did anyone figure out the fix?

**Comment by Junking1** [comment]
You can fix it with https://github.com/HKUST-Aerial-Robotics/Fast-Planner/pull/107
OR with  https://github.com/Junking1/Fast-Planner-for-ubuntu20.04

**Comment by dulex123** [comment]
@Junking1 It doesnt work for me on 20.04. It builds successfully and when I give it 2d nav goal it crashes? Any ideas?

**Comment by Junking1** [comment]
> @Junking1 It doesnt work for me on 20.04. It builds successfully and when I give it 2d nav goal it crashes? Any ideas?

Is it the same mistake as before?In fact, I suggest you download code from https://github.com/Junking1/Fast-Planner-for-ubuntu20.04  and try again.

**Comment by dulex123** [comment]
@Junking1 There are no errors. Exactly on setting 2D nav goal fast_planner node crashes. You have no issues? It works as expected?

**Comment by Junking1** [comment]
@dulex123 Yes, it works.Maybe you can output something in your nlopt library, making sure you're using the nlopt library you installed from the source code instead of the library from ros.

**Comment by ddefuijk** [comment]
@dulex123 I'm having the same problem. Did you find a solution for this?

**Comment by dulex123** [comment]
@ddefuijk nope. If you succeed please share the solution.

**Comment by ddefuijk** [comment]
@dulex123 It seems I made a typo while sourcing the workspace's setup.bash. I doubt that you have the same issue but you can check that.

**Comment by DeveloperFuloo** [comment]
crash in the same problem..  anyone fixed it?

**Comment by iamrajee** [comment]
> crash in the same problem.. anyone fixed it?

Stuck with the same issue for a long time now. Eventually, I had to shift to the melodic. There it works without any issue!

**Comment by iamrajee** [comment]
@eliabntt Your solution doesn't work for me. Has anybody else got it working?

**Comment by eliabntt** [comment]
Hi, do you get an error? I did exactly what I stated and it started working. Be sure to completely clean the build files (`catkin clean`/ `rm build devel logs -rf`)  after you removed the nlopt packages. Try also to remove the packages, reboot and install nlopt from source.

**Comment by iamrajee** [comment]
> catkin_make -DCMAKE_CXX_STANDARD=14

I tried that, but no luck :( Thanks anyway!

**Comment by eliabntt** [comment]
I don't see segfault here.

**Comment by qycqycqyc** [comment]
@eliabntt Thank you very much!
